src/vs/editor/contrib/colorPicker/browser/colorDetector.ts

```diff
@@ -142,16 +147,17 @@ export class ColorDetector extends Disposable implements IEditorContribution {
 		this._computePromise = createCancelablePromise(async token => {
 			const model = this._editor.getModel();
 			if (!model) {
-				return Promise.resolve([]);
+				return Promise.resolve({ colorData: [], usingDefaultDocumentColorProvider: false });
 			}
 			const sw = new StopWatch(false);
-			const colors = await getColors(this._languageFeaturesService.colorProvider, model, token);
+			const colorInfos = await getColors(this._languageFeaturesService.colorProvider, model, token);
```

Comment:
The `isDefaultColorDecoratorsEnabled` setting could be passed as argument to simplify this (and avoid computation whne turned off)

Reply:
The `getColors` function is also used in the StandaloneColorPickerParticipant, where the function is called independently of the setting